src/rviz/mesh_loader.cpp

```diff
@@ -441,6 +442,9 @@ void loadMaterials(const std::string& resource_path,
   {
     std::stringstream ss;
     ss << resource_path << "Material" << i;
+
+    // Prevent loading a material twice
+    if(Ogre::MaterialManager::getSingleton().getByName(ss.str()).get() != NULL) continue;
```

Comment:
`getByName` defaults to `ResourceGroupManager::AUTODETECT_RESOURCE_GROUP_NAME` I think. This piece of code is here because in my tests Ogre panics here saying that you cannot have resources with the same name. Needs more testing though.